client/state/notices/test/middleware.js

```diff
@@ -50,7 +53,7 @@ describe( 'middleware', () => {
 		test( 'should trigger the observer corresponding to the dispatched action type', () => {
 			noticesMiddleware( store )( noop )( dummyCreator( 'World' ) );
 
-			expect( store.dispatch ).to.have.been.calledWithMatch( {
+			sinon.assert.calledWithMatch( dispatchSpy, {
```

Comment:
Is there a reason you chose sinon over jest's `expect` which I think support partial matching?
I'm not saying to redo it with Jest! Just curious

Reply:
Both the original and the new assertions are Sinon. `expect( spy ).to.have.been.calledWithMatch()` comes from the `sinon-chai` plugin. `sinon.assert.calledWithMatch( spy )` comes directly from the `sinon` library.

Jest has most of Sinon functionality built-in (`jest.fn()`, `jest.spyOn`) and in the long term, we will probably abandon Sinon.